### DISCUSSION TITLE
For tags, switch from Unicode “letter” to “alphabetic” for allowed characters?

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/klog#367] [Feature ideas & discussions]
A “letter” (the allowed character class for [tags](https://github.com/jotaen/klog/blob/main/Specification.md#tag)) is currently defined as “character as defined by the Unicode Letter category (L)”.

For [[x]it! (a file format for check lists)](https://github.com/jotaen/xit), there was a discussion on HackerNews in which [one commentator suggested](https://news.ycombinator.com/item?id=32560102) that it would be better to use the Unicode “alphabetic” category instead, which appears to be a superset of “letter”.

This topic applies to klog in the same fashion, so depending of the outcome [of that investigation in [x]it!](https://github.com/jotaen/xit/issues/43), the definition of “letter” might need changing here as well.

Since “alphabetic” would be a superset of “letter”, this change would be backwards-compatible.

**Comment by jotaen** [comment]
tldr; I’m closing this for now, as I think the current definition is sufficient, especially considering that the “letter” definition is only relevant for tags, but not for any other text.

The Unicode categories are defined as:

- Letter: `Lu + Ll + Lt + Lm + Lo`
- Alphabetic: `Lu + Ll + Lt + Lm + Lo + Other_Lowercase + Other_Uppercase + Nl + Other_Alphabetic`

So, “alphabetic” is a superset of “letter”, that adds `Other_Lowercase + Other_Uppercase + Nl + Other_Alphabetic`:

- `Other_Lowercase` and `Other_Uppercase` contains a variety of miscellaneous characters such as roman numerals or letters/numbers with circles around it.
- `Letter_Number` contains “letterlike numeric characters”, mostly ancient ones, like roman, old persian, cuneiform, etc.
- `Other_Alphabetic` mostly contains code points from the “non-spacing mark” and “spacing combining marks” categories.

For tags, I’m not sure that any of these additional categories would be useful.
